webservices/legal_docs/archived_murs.py

```diff
@@ -0,0 +1,398 @@
+import json
+import datetime
+from json import JSONEncoder
+
+logger = logging.getLogger(__name__)
+LOCAL_DEBUG = False
+
+ALL_ARCHIVED_MURS = """
+    SELECT DISTINCT
+        mur_id,
+        mur_number as mur_no
+    FROM MUR_ARCH.ARCHIVED_MURS
+    ORDER BY mur_id
+"""
+
+SINGLE_MUR = """
+    SELECT DISTINCT
+        mur_number as mur_no,
+        mur_id,
+        mur_name,
+        open_date,
+        close_date
+    FROM MUR_ARCH.ARCHIVED_MURS
+    WHERE mur_number = %s
+"""
+
+MUR_COMPLAINANT = """
+    SELECT
+        complainant_respondent_code as code,
+        complainant_respondent_name as name
+    FROM MUR_ARCH.ARCHIVED_MURS
+    WHERE mur_id = %s AND complainant_respondent_code = 'C'
+    ORDER BY complainant_respondent_name
+"""
+
+MUR_RESPONDENT = """
+    SELECT
+        complainant_respondent_code as code,
+        complainant_respondent_name as name
+    FROM MUR_ARCH.ARCHIVED_MURS
+    WHERE mur_id = %s AND complainant_respondent_code = 'R'
+    ORDER BY complainant_respondent_name
+"""
+
+MUR_CITES = """
+    SELECT
+        citation as cite
+    FROM MUR_ARCH.ARCHIVED_MURS
+    WHERE mur_id = %s AND citation is not null
+    ORDER BY citation
+"""
+
+MUR_SUBJECT_LV1 = """
+    SELECT
+        mur,
+        level_1,
+        subject
+    FROM MUR_ARCH.SUBJCT_FROM_FILE
+    WHERE level_2::integer = 0 AND level_3::integer = 0
+    AND mur = %s
+"""
+
+MUR_SUBJECT_LV2 = """
+    SELECT
+        mur,
+        level_2,
+        subject
+    FROM MUR_ARCH.SUBJCT_FROM_FILE
+    WHERE level_2::integer > 0
+    AND level_3::integer = 0
+    AND mur = %s
+    AND level_1 = %s
+"""
+
+MUR_SUBJECT_LV3 = """
+    SELECT
+        mur,
+        level_3,
+        subject
+    FROM MUR_ARCH.SUBJCT_FROM_FILE
+    WHERE level_3::integer > 0
+    AND mur = %s
+    AND level_1 = %s
+    AND level_2 = %s
+"""
+
+MUR_DOCUMENTS = """
+    SELECT
+        document_id,
+        mur_no,
+        pdf_text,
+        mur_id,
+        length,
+        url
+    FROM MUR_ARCH.DOCUMENTS
+    WHERE mur_id = %s
+    ORDER BY document_id
+"""
+
+INSERT_DOCUMENT = """
+    INSERT INTO MUR_ARCH.DOCUMENTS(
+        document_id,
+        mur_no,
+        pdf_text,
+        mur_id,
+        length,
+        url
+    ) VALUES (:document_id, :mur_no, :pdf_text, :mur_id, :length, :url)
+"""
+
+
+# To display the open_date and close_date of JSON format inside object "mur"
+class DateTimeEncoder(JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, (datetime.date, datetime.datetime)):
+            return obj.isoformat()
+
+
+def load_archived_murs(mur_no=None):
+
+    """
+    Reads data for Archived MURs from a Postgres database (under schema:mur_arch),
+    assembles a JSON document corresponding to the mur, and indexes this document
+    in Elasticsearch in the index `archived_murs` with a doc_type of `murs`.
+    """
+    es = utils.get_elasticsearch_connection()
+    logger.info("Loading archived mur{0}...".format(mur_no))
+    if LOCAL_DEBUG:
```

Comment:
Where the `LOCAL_DEBUG` statements are the same as the logger.info statements, can we remove them? I would think they'd both print out when `LOCAL_DEBUG` is turned on.

Reply:
Yes, will do.Thanks